crypto/dh/dh_check.c

```diff
@@ -155,7 +155,7 @@ int DH_check(const DH *dh, int *ret)
     if (!DH_check_params(dh, ret))
         return 0;
 
-    ctx = BN_CTX_new();
+    ctx = BN_CTX_new_ex(dh->libctx);
```

Comment:
I assume you dont test this one because there is no path to it now.. except by calling it directly..

Reply:
There is a path to it - The EVP_PKEY_param_check() invokes it through DH_check_ex(). Actually the other change does not have any impact as in the quick check the libctx in the BN_CTX is not really used for anything.

I've verified that without this change the newly added test fails - due to failing EVP_PKEY_param_check().